python/lib/dependabot/python/file_parser/setup_file_parser.rb

```diff
@@ -1,79 +1,20 @@
 # frozen_string_literal: true
 
-require "dependabot/dependency"
-require "dependabot/errors"
-require "dependabot/file_parsers/base/dependency_set"
-require "dependabot/shared_helpers"
-require "dependabot/python/file_parser"
-require "dependabot/python/native_helpers"
-require "dependabot/python/name_normaliser"
+require_relative "setup_file_parser_base"
 
 module Dependabot
   module Python
     class FileParser
-      class SetupFileParser
+      class SetupFileParser < SetupFileParserBase
         INSTALL_REQUIRES_REGEX = /install_requires\s*=\s*\[/m.freeze
         SETUP_REQUIRES_REGEX = /setup_requires\s*=\s*\[/m.freeze
         TESTS_REQUIRE_REGEX = /tests_require\s*=\s*\[/m.freeze
         EXTRAS_REQUIRE_REGEX = /extras_require\s*=\s*\{/m.freeze
 
         CLOSING_BRACKET = { "[" => "]", "{" => "}" }.freeze
 
-        def initialize(dependency_files:)
-          @dependency_files = dependency_files
-        end
-
-        def dependency_set
-          dependencies = Dependabot::FileParsers::Base::DependencySet.new
-
-          parsed_setup_file.each do |dep|
-            # If a requirement has a `<` or `<=` marker then updating it is
-            # probably blocked. Ignore it.
-            next if dep["markers"].include?("<")
-
-            # If the requirement is our inserted version, ignore it
-            # (we wouldn't be able to update it)
-            next if dep["version"] == "0.0.1+dependabot"
-
-            dependencies <<
-              Dependency.new(
-                name: normalised_name(dep["name"], dep["extras"]),
-                version: dep["version"]&.include?("*") ? nil : dep["version"],
-                requirements: [{
-                  requirement: dep["requirement"],
-                  file: Pathname.new(dep["file"]).cleanpath.to_path,
-                  source: nil,
-                  groups: [dep["requirement_type"]]
-                }],
-                package_manager: "pip"
-              )
-          end
-          dependencies
-        end
-
         private
 
-        attr_reader :dependency_files
-
-        def parsed_setup_file
-          SharedHelpers.in_a_temporary_directory do
-            write_temporary_dependency_files
-
-            requirements = SharedHelpers.run_helper_subprocess(
-              command: "pyenv exec python #{NativeHelpers.python_helper_path}",
-              function: "parse_setup",
-              args: [Dir.pwd]
-            )
-
-            check_requirements(requirements)
-            requirements
-          end
-        rescue SharedHelpers::HelperSubprocessFailed => e
-          raise Dependabot::DependencyFileNotEvaluatable, e.message if e.message.start_with?("InstallationError")
-
-          parsed_sanitized_setup_file
```

Comment:
Following on from `jurre`s comment, if we move parsing both back into this class, we could do something like:

```
return [] unless setup_file

parsed_sanitized_setup_file
```